pandas/core/window/ewm.py

```diff
@@ -585,6 +588,19 @@ class ExponentialMovingWindowGroupby(BaseWindowGroupby, ExponentialMovingWindow)
 
     _attributes = ExponentialMovingWindow._attributes + BaseWindowGroupby._attributes
 
+    def __init__(self, obj, *args, _grouper=None, **kwargs):
+        super().__init__(obj, *args, _grouper=_grouper, **kwargs)
+
+        if not obj.empty and self.times is not None:
+            # sort the times and recalculate the deltas according to the groups
+            groupby_order = np.concatenate(list(self._grouper.indices.values())).astype(
+                np.int64
+            )
+            # error: Item "str" of "Union[str, ndarray, FrameOrSeries]" has no
+            # attribute "take"
+            self.times = self.times.take(groupby_order)
```

Comment:
Ideally we don't want to change this attribute too much since its still public to the user.

Instead could you define `_calculate_deltas(self.times.take(groupby_order), self.halflife)`